contracts/integrations/nevermined/MechFactorySubscription.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {AgentMechSubscription} from "./AgentMechSubscription.sol";
+
+/// @title Mech Factory Subscription - Periphery smart contract for managing subscription mech creation
+contract MechFactorySubscription {
+    event CreateSubscriptionMech(address indexed mech, uint256 indexed serviceId, uint256 minCreditsPerRequest,
+        address indexed subscriptionNFT, uint256 subscriptionTokenId);
+
+    // Agent factory version number
+    string public constant VERSION = "0.1.0";
+
+    /// @dev Registers service as a mech.
+    /// @param mechMarketplace Mech marketplace address.
+    /// @param serviceRegistry Service registry address.
+    /// @param serviceId Service id.
+    /// @param payload Mech creation payload.
+    /// @return mech The created mech instance address.
+    function createMech(
```

Comment:
Basically mech creation function as before, just depends on mech passed params to decode.